recipes/cpprestsdk/all/conanfile.py

```diff
@@ -55,7 +55,7 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        self.requires("boost/1.76.0")
+        self.requires("boost/1.77.0")
         self.requires("openssl/1.1.1k")
```

Comment:
Why not bump boost to 1.78.0?

(and openssl can be update to 1.1.1m in the meantime)

Reply:
That would be good in the long term. However the bump to applications using Boost 1.77.0 caused these many segfault/crash issues. These are now being resolved by these PRs. I'd rather not change anything further until we have stability again, if that's OK?

Reply:
Ok for me. But for openssl I think it's worth bumping to 1.1.1m

Reply:
That could be OK, although the [OpenSSL Release Notes](https://www.openssl.org/news/openssl-1.1.1-notes.html) say no major changes or CVE fixes between 1.1.1l and 1.1.1m, so not so critical as some OpenSSL releases.

Reply:
I understand.
The idea is to not introduce too much conflicts in versions required by CCI recipes, otherwise it can prevent builds in CI due to dependency graph resolution of conan (if a recipe depends on 2 other recipes but not openssl as a direct dependency, and these 2 dependencies depend on different versions of openssl in their recipe, then CI will fail).

Reply:
@garethsb you always can override boost to 1.78.0 (or another) from your profile

Reply:
@SSE4 "you always can override boost to 1.78.0 (or another) from your profile" - agreed, so no need to bump it in this recipe, right?
@SpaceIm, maybe I'm confused... I'm almost certain that Boost, cpprestsdk, websocketpp have all depended on different versions of OpenSSL at the same point in time more than once, and an app build doesn't fail, moreover it uses each of the binaries of those libraries, even a direct dependency on a specific version of OpenSSL doesn't change this, because of the default semver dependency mode?

Reply:
> @SpaceIm, maybe I'm confused... I'm almost certain that Boost, cpprestsdk, websocketpp have all depended on different versions of OpenSSL at the same point in time more than once, and an app build doesn't fail, moreover it uses each of the binaries of those libraries, even a direct dependency on a specific version of OpenSSL doesn't change this, because of the default semver dependency mode?

conan can accept only one version of a recipe in a dependency graph. Downstream recipes override versions of upstream. If there are two independant nodes in the dependency graph which have a different version, and no parent (root usually) can make the decision because it doesn't directly depend on openssl, then it fails (and the solution is to use `self.requires("openssl/1.1.1m", override=True)` in the root, but CI of CCI never does that)
So it depends of the dependency graph.

Reply:
Maybe let's not introduce the conflict? Only one deps needs openssl

https://github.com/conan-io/conan-center-index/blob/6fe6d5c259a66c771a797af5de68c2ec4fe0f5fb/recipes/websocketpp/all/conanfile.py#L33

And I suspect CCI would fail if we changed it here first

Reply:
> Maybe let's not introduce the conflict? Only one deps needs openssl
> 
> https://github.com/conan-io/conan-center-index/blob/6fe6d5c259a66c771a797af5de68c2ec4fe0f5fb/recipes/websocketpp/all/conanfile.py#L33
> 
> And I suspect CCI would fail if we changed it here first

No, because most recipes already moved to `openssl/1.1.1m`.

Reply:
I thought @prince-chrismc was right because CCI CI never builds missing packages. If we bump the version here first, then won't this PR fail because a version of websocketpp with the newer openssl dependency is not available?

Reply:
websocketpp won't fail because cpprestsdk will override the version, there is no conflict since cpprestsdk is the root of the dependency graph

Reply:
That's horribly confusing... I played with it locally and yeah your absolutely right https://github.com/conan-io/conan-center-index/pull/8967 since it works.

I guess there's a difference between `conan create recipes/cpprestsdk/all` and using a `conanfile` that's not very intuitive... Where in the conan main docs would be a good place to write this down? 

Thanks for sharing space... I always bumped the leaf dependencies first, I had no clue this was possible

Reply:
I guess I need to try it too, but does it still work if you don't build missing?

Reply:
Yep! if you create the package it's not required

(which make sense because installing, the overrides need to be in the root as well)

Reply:
Also, as I said in the discussion on Slack, the warning message displayed by Conan about overriding a dependency A's dependency B's version is **HORRIBLY** confusing, because it doesn't in fact mean that A will be rebuilt with that version of B, that depends on the package id mode, which is what led us to the Boost related segfaults and this sequence of PRs.